pkg/bot/slack_cloud.go

```diff
@@ -161,6 +160,7 @@ func withRetries(ctx context.Context, log logrus.FieldLogger, maxRetries int, fn
 }
 
 func (b *CloudSlack) start(ctx context.Context) error {
+	b.messages = make(chan *pb.ConnectResponse, platformMessageChannelSize)
 	defer b.shutdown()
```

Comment:
WDYT about scoping this? 
```suggestion
	messages := make(chan *pb.ConnectResponse, platformMessageChannelSize)
	defer b.shutdown(messages)
```

This way it will be easier to track. I tested, and it can be easily shared only with `startMessageProcessor` and `shutdown` just via input arg. For me it will be a bit cleaner 🙂